src/stream/src/executor/backfill.rs

```diff
@@ -139,11 +139,12 @@ where
             state_table.init_epoch(first_barrier.epoch);
         }
 
-        // Maintain backwards compatibility with no state table.
         let is_finished = if let Some(state_table) = self.state_table.as_mut() {
             Self::check_all_vnode_finished(state_table, state_len).await?
+                || !first_barrier.is_newly_added(self.actor_id)
```

Comment:
Adding this to be conservative.

Reply:
Could you please change it to an assert? I think it will be helpful to help us to find inconsistencies quickly.